components/usage/pkg/apiv1/usage_test.go

```diff
@@ -222,6 +222,31 @@ func TestReconcile(t *testing.T) {
 		}))
 		require.EqualValues(t, expectedUsage, updates[0])
 	})
+
+	t.Run("handles instances without stopping but stopped time", func(t *testing.T) {
+		instance := db.WorkspaceInstanceForUsage{
+			ID:          uuid.New(),
+			WorkspaceID: dbtest.GenerateWorkspaceID(),
+			OwnerID:     uuid.New(),
+			ProjectID: sql.NullString{
+				String: "my-project",
+				Valid:  true,
+			},
+			WorkspaceClass:     db.WorkspaceClass_Default,
+			Type:               db.WorkspaceType_Regular,
+			UsageAttributionID: db.NewTeamAttributionID(uuid.New().String()),
+			StartedTime:        db.NewVarCharTime(now.Add(1 * time.Minute)),
+			StoppedTime:        db.NewVarCharTime(now.Add(2 * time.Minute)),
+		}
+
+		inserts, updates, err := reconcileUsage([]db.WorkspaceInstanceForUsage{instance}, []db.Usage{}, pricer, now)
+		require.NoError(t, err)
+		require.Len(t, inserts, 1)
+		require.Len(t, updates, 0)
+
+		require.EqualValues(t, db.NewCreditCents(0.17), inserts[0].CreditCents)
+		require.EqualValues(t, instance.StoppedTime, inserts[0].EffectiveTime)
```

Comment:
Unsure how this test passes, given we're not including the select in https://github.com/gitpod-io/gitpod/pull/15191/files#r1041072396

Reply:
it doesn't persist or read from the db

Reply:
Ah, that would be why. Thanks!